src/display/api.js

```diff
@@ -1435,6 +1436,13 @@ var WorkerTransport = (function WorkerTransportClosure() {
       this.destroyed = true;
       this.destroyCapability = createPromiseCapability();
 
+      if (this._passwordCapability) {
+        var err = new Error('Worker was destroyed during onPassword callback');
+        this.loadingTask._capability.reject(err);
```

Comment:
*This is definitely getting more complicated than I imagined; thanks for your in-depth review comments thus far!*

I tried removing that, but it appears to be causing some problems:
When calling `loadingTask.destroy();` in the `onPassword` handler, it seems that the worker might be terminated *before* it has had a chance to send the (new) `PasswordException` (as outlined in comment https://github.com/mozilla/pdf.js/pull/7926/files#r94335620).

The `loadingTask` will thus remain in a pending state, without ever getting either resolved *or* rejected. Note that I'm seeing this both in manual tests in the viewer, as well as in the unit-tests.

How would you propose that we handle this case?
Could/should we use the `WorkerTask` (in worker.js) to track the password promise there, to ensure that worker termination waits for the rejected `_passwordCapability` from api.js?